src/Utils/util.test.ts

```diff
@@ -70,4 +71,221 @@ describe('util', () => {
             expect(actual).toBe(true)
         })
     })
+
+    describe('isActionUniqueTEXT', () => {
+        it('exercise TEXT action creation', () => {
```

Comment:
This looks like the only `it` block in the `describe`

I also didn't know it would allow to put `test` wthin an `it`...  

Your tests seems to be structured in a weird way. Try it like this:

```
describe('isActionUnique', () => {
  describe('TEXT actions', () => {
    test('as first action created', () => { ... })
  })

  describe(`ENDSESSION actions`, () => { ... })
  describe(`CARD actions`, () => { ... })
})
```

This poses other question. It seems like we're doing the same tests for different types of actions but the `isActionUnique` doesn't do anything different based on the type. What do these tests cover that the previous didn't?

Reply:
Realigned the test to use the nested describes, and lose the it wrappers.

Speaking to isActionUnique itself, the code stringifies most of the action object, including the actionType property, which varies between action types. The product currently has a limited trap on adding duplicate ENDSESSION actions for example. Now, using this code, the duplicate action detection is significantly stronger.